File: app/src/main/kotlin/com/mobileappdev/homeworkplanner/AssignmentList.java
package com.mobileappdev.homeworkplanner;

import android.content.Context;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.List;

public class AssignmentList {
    private static AssignmentList sAssignmentList;

    private List<Assignment> mAssignments;

    public static AssignmentList get(Context context){
        if (sAssignmentList == null){
            sAssignmentList = new AssignmentList(context);
        }
        return sAssignmentList;
    }

    private AssignmentList(Context context){
        mAssignments = new ArrayList<>();

        // TODO: Populate with all assignments from database
    }

    public List<Assignment> getAssignments(){ return mAssignments; }
}

File: app/src/main/kotlin/com/mobileappdev/homeworkplanner/AssignmentList.kt
package com.mobileappdev.homeworkplanner

import java.util.ArrayList

object AssignmentList {
    var mAssignments: ArrayList<Assignment> = ArrayList()

    fun createAssignment(dataMap: Map<String, Any>, documentId: String) {
        val assignment = Assignment()
        assignment.name = dataMap["assignmentName"] as String?
        assignment.dueDate = dataMap["dueDate"] as String?
        assignment.dueTime = dataMap["dueTime"] as String?
        assignment.importance = dataMap["importance"] as String?
        assignment.parentClass = dataMap["className"] as String?
        assignment.startDate = dataMap["startDate"] as String?
        assignment.timeEstimate = dataMap["timeEstimate"] as Long
        assignment.actualTimeSpent = dataMap["timeSpent"] as Long
        assignment.isComplete = dataMap["isComplete"] as Boolean?
        assignment.documentId = documentId
        mAssignments.add(assignment)
    }
}
